### PR TITLE
gh-121300: Add `replace` to `copy.__all__`

### PR DESCRIPTION
**Pull request by max-muoto**
Adds missing replace to `copy.__all__`

* Issue: gh-121300

### CODE REVIEW

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

File: Lib/copy.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ class Error(Exception):
     pass
 error = Error   # backward compatibility
 
-__all__ = ["Error", "copy", "deepcopy"]
+__all__ = ["Error", "copy", "deepcopy", "replace"]
```
Comment (pygeek) [inline Lib/copy.py:60]: Consider adding a regression test so this doesn't happen again. See helper method `support.test__all__` in the test directory—it automatically detects public methods based on whether they follow the public name convention.

Comment (max-muoto) [reply Lib/copy.py:60]: Sure, let me go ahead, good callout.

**Comment by max-muoto** [comment]
@serhiy-storchaka If you want to take another look.

**Review by Eclips4 — APPROVED** [review]
Thanks!

**Review by kumaraditya303 — APPROVED** [review]

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**